client/src/commonMain/kotlin/com/algolia/search/model/indexing/Partial.kt

```diff
@@ -180,8 +184,16 @@ public sealed class Partial {
             val element = decoder.asJsonInput().jsonObject
             val key = element.keys.first()
             val attribute = key.toAttribute()
-            val operation = element.getValue(key).jsonObject[Key_Operation]?.jsonPrimitiveOrNull?.content
-            val jsonElement = element.getValue(key).jsonObject.getValue(KeyValue)
+            val value = element.getValue(key)
+            val hasOperation = (value is JsonObject && value.jsonObject.containsKey(Key_Operation))
+            val (operation, jsonElement) = if (hasOperation) {
+                Pair(
+                    value.jsonObject[Key_Operation]?.jsonPrimitiveOrNull?.content,
+                    value.jsonObject.getValue((KeyValue))
+                )
+            } else {
+                Pair(null, value)
+            }
```

Comment:
```suggestion
            val operation = if (hasOperation) value.jsonObject[Key_Operation]?.jsonPrimitiveOrNull?.content else null
            val jsonElement = if (hasOperation) value.jsonObject.getValue((KeyValue)) else value
```
_nitpick: we can avoid an extra object creation, and increase readability_